File: testdata/minimal-testbeds/dart/lib/greeter.dart
import 'helpers.dart';
import 'tone.dart';
export 'tone.dart';

mixin Auditable {
  String audit(String s) => format(s);
}

class Greeter with Auditable {
  final Tone tone;

  Greeter({this.tone = Tone.casual});

  Greeter.formal() : tone = Tone.formal;

  String greet(String name) {
    return format(name);
  }

  String greetWithTone(String name) {
    final base = greet(name);
    return decorate(base, tone);
  }

  String get label => 'greeter';
}

/// Top-level helper — must stay parentless (not sticky Greeter ParentID).
String shout(String name) {
  return format(name);
}

File: testdata/minimal-testbeds/zig/src/greeter.zig
const helpers = @import("helpers.zig");

pub const NameSlice = []const u8;

pub const Tone = enum {
    casual,
    formal,
};

pub const Greeter = struct {
    tone: Tone,

    pub fn greet(self: Greeter, name: []const u8) []const u8 {
        _ = self;
        return format(name);
    }

    pub fn shout(self: Greeter, name: []const u8) []const u8 {
        _ = self;
        return helpers.upper(format(name));
    }

    pub const Stats = struct {
        count: usize,
    };
};

pub fn format(s: []const u8) []const u8 {
    return s;
}
